modules/analysis-common/src/test/java/org/elasticsearch/analysis/common/CommonAnalysisFactoryTests.java

```diff
@@ -46,7 +51,31 @@
 
     @Override
     protected Map<String, Class<?>> getCharFilters() {
-        Map<String, Class<?>> filters = new HashMap<>(super.getCharFilters());
+        Map<String, Class<?>> filters = new TreeMap<>(super.getCharFilters());
+        return filters;
+    }
+
+    @Override
+    protected Map<String, Class<?>> getPreBuiltTokenFilters() {
+        Map<String, Class<?>> filters = new TreeMap<>(super.getPreBuiltTokenFilters());
+        filters.put("asciifolding", null);
```

Comment:
I've sort of preserved this `null` behavior from the original test case. It is a bit funky and magical and I'm tempted to remove it but I've kept it for now to get opinions on it.